### PR TITLE
Allow user to use default image on failure

### LINKED ISSUE #1788: Allow skipping Image build failure
When I start a workspace and the image build fails, there is currently no way to start a workspace on that commit.
I'd like to allow users to start a workspace using the default image.

### PR DESCRIPTION
**Pull request by JesterOrNot**
Fixes #1788

### CODE REVIEW

**Comment by JesterOrNot** [comment]
Hmm out of space

File: components/dashboard/src/components/start-workspace.tsx
Comment (svenefftinge) [inline components/dashboard/src/components/start-workspace.tsx:360]: We'd need to pass an argument to the server API, telling it to ignore the image configuration and use the default.

**Comment by JesterOrNot** [comment]
ok so it's a conditional logic error

**Comment by JesterOrNot** [comment]
Finally.

**Comment by JesterOrNot** [comment]
1. To test this PR open the following workspace
[Open in Gitpod](http://sh-skip-build-fail.staging.gitpod-dev.com/#https://github.com/JesterOrNot/Playground/tree/sh/build-fail)
2. See it fails
3. Click open default workspace
4. See a new workspace is created on the commit with the default image.

File: components/server/src/workspace/workspace-starter.ts
Comment (corneliusludmann) [inline components/server/src/workspace/workspace-starter.ts:67]: Could we use here the workspace image configured in the `values.yaml`? https://github.com/gitpod-io/gitpod/blob/3b1b7f11183c4972bd08478a3e44b183edf5ada1/chart/values.yaml#L255-L258

See e.g. here: https://github.com/gitpod-io/gitpod/blob/3b1b7f11183c4972bd08478a3e44b183edf5ada1/components/server/src/workspace/config-provider.ts#L284

Comment (csweichel) [inline components/server/src/workspace/workspace-starter.ts:67]: Please use [`ConfigProvider.DEFAULT_IMAGE`](https://github.com/gitpod-io/gitpod/blob/9853db2c44d8fd921ec8b9b81553a7ec48efe4d2/components/server/src/workspace/config-provider.ts#L284) instead

File: components/dashboard/src/components/start-workspace.tsx
Comment (svenefftinge) [inline components/dashboard/src/components/start-workspace.tsx:124]: any reason to write the type here?

File: components/gitpod-protocol/src/gitpod-service.ts
Comment (svenefftinge) [inline components/gitpod-protocol/src/gitpod-service.ts:59]: let's introduce `options` object

File: components/server/src/workspace/workspace-starter.ts
Comment (svenefftinge) [inline components/server/src/workspace/workspace-starter.ts:48]: we should switch to an `options` arg here, too.

Comment (svenefftinge) [reply components/server/src/workspace/workspace-starter.ts:48]: the name should be `forceDefaultImage`

File: components/dashboard/src/components/start-workspace.tsx
Comment (JesterOrNot) [reply components/dashboard/src/components/start-workspace.tsx:124]: Yes, Theia shows it as an error.
`Parameter 'workspaceStartedResult' implicitly has an 'any' type.ts(7006)`

**Comment by JesterOrNot** [comment]
@svenefftinge take another look?

File: components/dashboard/src/components/start-workspace.tsx
Comment (svenefftinge) [inline components/dashboard/src/components/start-workspace.tsx:359]: This button should only be shown when an image build failed or an image is not available.
The label could be "Start with Default Docker Image".

**Comment by svenefftinge** [comment]
Nice! It should replace the existing button, though.
Currently it looks like this:

Screenshot 2020-09-08 at 16 36 16

File: components/dashboard/src/components/start-workspace.tsx
Comment (svenefftinge) [inline components/dashboard/src/components/start-workspace.tsx:356]: use `isBuildingWorkspaceImage` instead

Comment (JesterOrNot) [reply components/dashboard/src/components/start-workspace.tsx]: Done.

File: components/server/src/workspace/gitpod-server-impl.ts
Comment (csweichel) [inline components/server/src/workspace/gitpod-server-impl.ts:372]: here you should explicitly set the options that you pass to `startWorkspace`. Since this is coming from the websocket interface, you don't really know what you're passing along (could be way more than the type, or nothing like the type at all - also see https://www.youtube.com/watch?v=oGeEoaplMWA).

```suggestion
            return await this.workspaceStarter.startWorkspace({ span }, workspace, user, await envVars, {
                forceDefaultImage: !!options.forceDefaultImage
            });
```

**Review by csweichel — COMMENTED** [review]
Works like a charm, and other than the comment below code LGTM.
Once fixed I'll approve.

**Review by csweichel — APPROVED** [review]
Awesome. Thanks for the fix 👍

**Comment by JesterOrNot** [comment]
Thanks, will merge after CI passes :rocket:

**Pull request merged**
